src/core/structure/sheet.ts

```diff
@@ -396,18 +435,114 @@ export default class Sheet {
     return this.cellData.get(cellKey)!;
   }
 
-  private resolveCell(cell: Cell): boolean {
-    const value = this.expressionHandler.evaluate(cell.formula);
-    if (value == null) {
-      cell.state = CellState.INVALID_EXPRESSION;
-      return false;
+  /**
+   * Returns true if the value of the cell has changed.
+   */
+  private resolveCell(cell: Cell, colKey: ColumnKey, rowKey: RowKey): boolean {
```

Comment:
This function is internally emitting events when handling other cells reference so I would say:
- Your comments is underplaying the significance of this function, I would prefer if it is removed. I'm not a fan of comments
- There is no need to return true/false when changes happen rather, I would suggest emitting the events directly from this function because you are already doing that inside of this function when updating other cells

Reply:
I would prefer to leave a comment describing the return value when its purpose isn't immediately obvious.
> There is no need to return true/false when changes happen rather, I would suggest emitting the events directly from 

I'll move the event emission logic into `resolveCell`; I agree it makes more sense there. The return value is still required though, since referring cells aren't affected if the cell's value doesn't change.